R/antigen.garnish_io.R

```diff
@@ -275,8 +275,11 @@ garnish_variants <- function(vcfs, intersect = TRUE){
 
   # check that VCF is SnpEff-annotated
 
-    if (!vdt[, INFO %>%
-      stringr::str_detect(fixed("ANN=")) %>% all])
+    if (
+        vdt[, INFO %>% unique] %>% is.na ||
+        !vdt[, INFO %>%
+      stringr::str_detect(fixed("ANN=")) %>% all]
+      )
       stop(paste0("\nInput file \n", vcfs[ivf], "\nis missing INFO SnpEff annotations"))
```

Comment:
namespace before stringr::fixed here?
first evaluates if any INFO field is.na, only first element will be evaluated by || if that one is NA then it triggers, do we want it to be is.na %>% any?  the first one shouldnt be particularly special and all variants should have an INFO field that is non-NA.  Plus second one is checking at they all have "ANN=" so i assume you want  %>% any in the first argument to ||.

Reply:
added fixed. `vdt[, INFO %>% unique] %>% is.na` is needed because if the INFO column exists but is NA, str_detect will fail